src/Team-Services-Bot.Api/Dialogs/ConnectDialog.cs

```diff
@@ -61,49 +63,124 @@ public Task StartAsync(IDialogContext context)
             return Task.CompletedTask;
         }
 
-        private async Task MessageReceivedAsync(IDialogContext context, IAwaitable<IMessageActivity> result)
+        private static string GeneratePin()
         {
-            var activity = await result;
+            var generator = new Random();
```

Comment:
I think a security reviewer would not be very comfortable with using Random(), there are better ways to generate cryptographically random numbers. My friendly local security person tells me I should use System.Security.Cryptography.RNGCryptoServiceProvider.